transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/session/GatewaySessionCtx.java

```diff
@@ -97,6 +101,10 @@ private void onDeviceConnect(String deviceName, String deviceType) {
                 relationService.saveRelationAsync(new EntityRelation(gateway.getId(), newDevice.getId(), "Created"));
                 return newDevice;
             });
+            if (null == device) {
+                log.debug("[{}] NOT allowed add to add device [{}] to the gateway session", gatewaySessionId, deviceName);
```

Comment:
please change this to warning. Also, the data upload request will be failed, but gateway can't know if it is doing something wrong. So, the best idea is to throw RuntimeException or AdaptorException and close gateway session in the MqttTransportHandler.handleMqttPublishMsg